cogs/music.py

```diff
@@ -0,0 +1,699 @@
+        return user != self.bot.user and reaction.message.id == embed_msg.id and str(reaction) in QUEUE_ACTIONS
+
+    def resume_playing(self, ctx):
+        ctx.voice_client.resume()
+        self.track_start_time += time.perf_counter() - self.pause_start
+        self.pause_start = None
+
+    def compute_curr_time(self, paused: bool) -> float:
+        return (self.pause_start if paused else time.perf_counter()) - self.track_start_time
+
+    def total_queue(self) -> Iterable:
+        return chain(self.track_queue, self.track_history) if self.looping_queue else self.track_queue
+
+    def total_len(self) -> int:
+        return len(self.track_queue) + len(self.track_history) if self.looping_queue else len(self.track_queue)
+
+    def play_track(self, ctx, *, skip_str: Optional[str] = None, delta: int = 0):
+        ctx.voice_client.play(
+            discord.PCMVolumeTransformer(
+                discord.FFmpegPCMAudio(
+                    self.playing[0]["url"],
+                    before_options=FFMPEG_BEFORE_OPTS if skip_str is None else f"{FFMPEG_BEFORE_OPTS} -ss {skip_str}",
+                    options=f'-filter:a "{self.speed_flag}" {FFMPEG_OPTS}',
+                ),
+                self.volume_level / 100,
+            ),
+            after=self.release_lock,
+        )
+        self.track_start_time = time.perf_counter() - delta
+
+    def from_total(self, idx):
+        q_len: int = self.total_len()
+        if idx >= q_len or idx < -q_len:
+            return None
+        idx = idx % q_len
+        return (
+            (self.track_queue, idx)
+            if (idx < len(self.track_queue))
+            else (self.track_history, idx - len(self.track_queue))
+        )
+
+    def subc_decision(self, subc: str) -> tuple[Callable, Optional[Callable]]:
+        match subc:
+            case "play":
+                return (self.play, lambda x: x)
+            case "playback_speed" | "ps" | "speed":
+                return (self.playback_speed, conv_arg(float, True))
+            case "goto_time" | "gt":
+                return (self.goto_time, conv_arg(lambda x: x, True))
+            case "forward_time" | "ft":
+                return (self.forward_time, conv_arg(lambda x: x, True))
+            case "backwards_time" | "bt" | "rewind":
+                return (self.backwards_time, conv_arg(lambda x: "30" if x is None else x, False))
+            case "loop":
+                return (self.loop, conv_arg(lambda x: x, True))
+            case "print_queue" | "pq":
+                return (self.print_queue, conv_arg(lambda x: 0 if x is None else int(x), False))
+            case "status" | "now_playing" | "np":
+                return (self.music_status, None)
+            case "remove" | "pop":
+                return (self.remove_track, conv_arg(int, True))
+            case "insert":
+                return (self.insert_track, conv_arg(lambda x: (lambda y, z: (int(y), z))(*x.split(maxsplit=1)), True))
```

Comment:
define a named function with some comments to explain this one rather than a one-liner as originally discussed